@christophermay07 : The server gets the following error "[ERROR] Caused by: java.lang.NoClassDefFoundError: javassist/util/proxy/MethodFilter"  which causes ValidationSuite fails on Github Pull Request Validator job.  We will take a look at further.

--- a/Product/Production/Adapters/PatientDiscovery_a0/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/adapter/deferred/request/queue/AdapterPatientDiscoveryDeferredReqQueueProcessTest.java
+++ b/Product/Production/Adapters/PatientDiscovery_a0/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/adapter/deferred/request/queue/AdapterPatientDiscoveryDeferredReqQueueProcessTest.java
@@ -71,8 +71,8 @@ public class AdapterPatientDiscoveryDeferredReqQueueProcessTest {
         context.checking(new Expectations() {
             {
                 oneOf(mockImpl).processPatientDiscoveryDeferredReqQueue(
-                        with(any(PatientDiscoveryDeferredReqQueueProcessRequestType.class)),
-                        with(any(javax.xml.ws.WebServiceContext.class)));
+                        with(aNull(PatientDiscoveryDeferredReqQueueProcessRequestType.class)),
+                        with(aNull(javax.xml.ws.WebServiceContext.class)));
                 will(returnValue(new PatientDiscoveryDeferredReqQueueProcessResponseType()));
             }
         });

--- a/Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/async/AsyncMessageProcessHelperTest.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/async/AsyncMessageProcessHelperTest.java
@@ -188,7 +188,7 @@ public class AsyncMessageProcessHelperTest {
         context.checking(new Expectations() {
             {
                 exactly(1).of(mockDao).save(with(any(AsyncMsgRecord.class)));
-                exactly(1).of(mockDao).queryByMessageIdAndDirection(with(any(String.class)), with(any(String.class)));
+                exactly(1).of(mockDao).queryByMessageIdAndDirection(with(any(String.class)), with(aNull(String.class)));
                 will(returnValue(returnMockRecords()));
             }
         });

--- a/Product/Production/Deploy/pom.xml
+++ b/Product/Production/Deploy/pom.xml
@@ -10,10 +10,11 @@
   <name>CONNECT Deployment</name>
   <packaging>pom</packaging>
   <modelVersion>4.0.0</modelVersion>
+  
   <modules>
     <module>jboss7</module>
-    </modules>
-  
+  </modules>
+
   <reporting>
     <plugins>
       <plugin>
@@ -28,7 +29,6 @@
   </reporting>
 
   <dependencies>
-
     <!-- Security -->
     <dependency>
       <groupId>org.owasp.esapi</groupId>
@@ -694,7 +694,7 @@
           <artifactId>asm-attrs</artifactId>
         </dependency>
         <dependency>
-          <groupId>javassist</groupId>
+          <groupId>org.javassist</groupId>
           <artifactId>javassist</artifactId>
         </dependency>
       </dependencies>
@@ -811,22 +811,24 @@
         <module>weblogic</module>
       </modules>
     </profile>
-  <!-- JBOSS7 -->
+
+    <!-- JBoss 7 -->
     <profile>
       <id>jboss7</id>
-    <activation>
+      <activation>
         <activeByDefault>true</activeByDefault>
       </activation>
-    <modules>
-    <module>jboss7</module>
-    </modules>
-  </profile>
-    <!-- Glassfish -->
+      <modules>
+        <module>jboss7</module>
+      </modules>
+    </profile>
+
+    <!-- GlassFish -->
     <profile>
       <id>glassfish</id>
-    <modules>
-    <module>glassfish</module>
-    </modules>
-  </profile>
+      <modules>
+        <module>glassfish</module>
+      </modules>
+    </profile>
   </profiles>
 </project>

--- a/Product/Production/Gateway/CONNECTGatewayWeb/src/test/java/gov/hhs/fha/nhinc/patientcorrelation/nhinc/PatientCorrelationServiceSecuredTest.java
+++ b/Product/Production/Gateway/CONNECTGatewayWeb/src/test/java/gov/hhs/fha/nhinc/patientcorrelation/nhinc/PatientCorrelationServiceSecuredTest.java
@@ -65,7 +65,7 @@ public class PatientCorrelationServiceSecuredTest {
                 oneOf(mockFactory).createPatientCorrelationService();
                 will(returnValue(mockService));
 
-                oneOf(mockService).addPatientCorrelation(with(same(request)), with(any(AssertionType.class)));
+                oneOf(mockService).addPatientCorrelation(with(same(request)), with(aNull(AssertionType.class)));
                 will(returnValue(expectedResponse));
 
             }
@@ -89,7 +89,7 @@ public class PatientCorrelationServiceSecuredTest {
                 oneOf(mockFactory).createPatientCorrelationService();
                 will(returnValue(mockService));
 
-                oneOf(mockService).retrievePatientCorrelations(with(same(request)), with(any(AssertionType.class)));
+                oneOf(mockService).retrievePatientCorrelations(with(same(request)), with(aNull(AssertionType.class)));
                 will(returnValue(expectedResponse));
 
             }

--- a/Product/Production/Gateway/CONNECTGatewayWeb/src/test/java/gov/hhs/fha/nhinc/patientcorrelation/nhinc/PatientCorrelationServiceUnsecuredTest.java
+++ b/Product/Production/Gateway/CONNECTGatewayWeb/src/test/java/gov/hhs/fha/nhinc/patientcorrelation/nhinc/PatientCorrelationServiceUnsecuredTest.java
@@ -63,7 +63,7 @@ public class PatientCorrelationServiceUnsecuredTest {
                 oneOf(mockFactory).createPatientCorrelationService();
                 will(returnValue(mockService));
 
-                oneOf(mockService).addPatientCorrelation(with(same(request)), with(any(AssertionType.class)));
+                oneOf(mockService).addPatientCorrelation(with(same(request)), with(aNull(AssertionType.class)));
                 will(returnValue(expectedResponse));
 
             }
@@ -85,7 +85,7 @@ public class PatientCorrelationServiceUnsecuredTest {
                 oneOf(mockFactory).createPatientCorrelationService();
                 will(returnValue(mockService));
 
-                oneOf(mockService).retrievePatientCorrelations(with(same(request)), with(any(AssertionType.class)));
+                oneOf(mockService).retrievePatientCorrelations(with(same(request)), with(aNull(AssertionType.class)));
                 will(returnValue(expectedResponse));
 
             }

--- a/Product/Production/Services/AdminDistributionCore/src/test/java/gov/hhs/fha/nhinc/admindistribution/outbound/StandardOutboundAdminDistributionTest.java
+++ b/Product/Production/Services/AdminDistributionCore/src/test/java/gov/hhs/fha/nhinc/admindistribution/outbound/StandardOutboundAdminDistributionTest.java
@@ -121,7 +121,7 @@ public class StandardOutboundAdminDistributionTest {
         context.checking(new Expectations() {
             {
                 oneOf(mockPolicyCheck).checkOutgoingPolicy(with(any(RespondingGatewaySendAlertMessageType.class)),
-                        with(any(String.class)));
+                        with(aNull(String.class)));
                 will(returnValue(allow));
             }
         });

--- a/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/OutboundDocSubmissionStrategyTest.java
+++ b/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/OutboundDocSubmissionStrategyTest.java
@@ -89,8 +89,8 @@ public class OutboundDocSubmissionStrategyTest {
     private void expectMockProxyInvokeWithG0() {
         context.checking(new Expectations() {
             {
-                oneOf(mockProxy).provideAndRegisterDocumentSetB(with(any(ProvideAndRegisterDocumentSetRequestType.class)),
-                        with(any(AssertionType.class)), with(any(NhinTargetSystemType.class)), with(equal(NhincConstants.GATEWAY_API_LEVEL.LEVEL_g0)));
+                oneOf(mockProxy).provideAndRegisterDocumentSetB(with(aNull(ProvideAndRegisterDocumentSetRequestType.class)),
+                        with(aNull(AssertionType.class)), with(aNull(NhinTargetSystemType.class)), with(equal(NhincConstants.GATEWAY_API_LEVEL.LEVEL_g0)));
                 will(returnValue(createRegistryResponseType()));
             }
         });
@@ -99,8 +99,8 @@ public class OutboundDocSubmissionStrategyTest {
     private void expectMockProxyInvokeWithG1() {
         context.checking(new Expectations() {
             {
-                oneOf(mockProxy).provideAndRegisterDocumentSetB(with(any(ProvideAndRegisterDocumentSetRequestType.class)),
-                        with(any(AssertionType.class)), with(any(NhinTargetSystemType.class)), with(equal(NhincConstants.GATEWAY_API_LEVEL.LEVEL_g1)));
+                oneOf(mockProxy).provideAndRegisterDocumentSetB(with(aNull(ProvideAndRegisterDocumentSetRequestType.class)),
+                        with(aNull(AssertionType.class)), with(aNull(NhinTargetSystemType.class)), with(equal(NhincConstants.GATEWAY_API_LEVEL.LEVEL_g1)));
                 will(returnValue(createRegistryResponseType()));
             }
         });

--- a/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/OutboundDocSubmissionDeferredRequestStrategyTest.java
+++ b/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/OutboundDocSubmissionDeferredRequestStrategyTest.java
@@ -94,8 +94,8 @@ public class OutboundDocSubmissionDeferredRequestStrategyTest {
         context.checking(new Expectations() {
             {
                 oneOf(mockProxy11).provideAndRegisterDocumentSetBRequest11(
-                        with(any(ProvideAndRegisterDocumentSetRequestType.class)), with(any(AssertionType.class)),
-                        with(any(NhinTargetSystemType.class)));
+                        with(aNull(ProvideAndRegisterDocumentSetRequestType.class)), with(aNull(AssertionType.class)),
+                        with(aNull(NhinTargetSystemType.class)));
                 will(returnValue(createXDRAcknowledgementType()));
             }
         });
@@ -105,8 +105,8 @@ public class OutboundDocSubmissionDeferredRequestStrategyTest {
         context.checking(new Expectations() {
             {
                 oneOf(mockProxy20).provideAndRegisterDocumentSetBRequest20(
-                        with(any(ProvideAndRegisterDocumentSetRequestType.class)), with(any(AssertionType.class)),
-                        with(any(NhinTargetSystemType.class)));
+                        with(aNull(ProvideAndRegisterDocumentSetRequestType.class)), with(aNull(AssertionType.class)),
+                        with(aNull(NhinTargetSystemType.class)));
                 will(returnValue(createRegistryResponseType()));
             }
         });

--- a/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/OutboundDocSubmissionDeferredResponseStrategyTest.java
+++ b/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/OutboundDocSubmissionDeferredResponseStrategyTest.java
@@ -100,8 +100,8 @@ public class OutboundDocSubmissionDeferredResponseStrategyTest {
         context.checking(new Expectations() {
             {
                 oneOf(mockProxy11).provideAndRegisterDocumentSetBDeferredResponse11(
-                        with(any(RegistryResponseType.class)), with(any(AssertionType.class)),
-                        with(any(NhinTargetSystemType.class)));
+                        with(aNull(RegistryResponseType.class)), with(aNull(AssertionType.class)),
+                        with(aNull(NhinTargetSystemType.class)));
                 will(returnValue(createXDRAcknowledgementType()));
             }
         });
@@ -111,8 +111,8 @@ public class OutboundDocSubmissionDeferredResponseStrategyTest {
         context.checking(new Expectations() {
             {
                 oneOf(mockProxy20).provideAndRegisterDocumentSetBDeferredResponse20(
-                        with(any(RegistryResponseType.class)), with(any(AssertionType.class)),
-                        with(any(NhinTargetSystemType.class)));
+                        with(aNull(RegistryResponseType.class)), with(aNull(AssertionType.class)),
+                        with(aNull(NhinTargetSystemType.class)));
                 will(returnValue(createRegistryResponseType()));
             }
         });

--- a/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyImplTest.java
+++ b/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyImplTest.java
@@ -79,8 +79,9 @@ public class EntityDocSubmissionProxyImplTest {
     private void setEntityOrchJavaImplExpectations() {
         context.checking(new Expectations() {
             {
-                oneOf(mockOutboundDS).provideAndRegisterDocumentSetB(with(any(ProvideAndRegisterDocumentSetRequestType.class)), with(any(AssertionType.class)),
-                        with(any(NhinTargetCommunitiesType.class)), with(any(UrlInfoType.class)));
+                oneOf(mockOutboundDS).provideAndRegisterDocumentSetB(
+                    with(aNull(ProvideAndRegisterDocumentSetRequestType.class)), with(aNull(AssertionType.class)),
+                    with(aNull(NhinTargetCommunitiesType.class)), with(aNull(UrlInfoType.class)));
                 will(returnValue(createRegistryResponseType()));
             }
         });

--- a/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceSecuredImplTest.java
+++ b/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceSecuredImplTest.java
@@ -104,9 +104,10 @@ public class EntityDocSubmissionProxyWebServiceSecuredImplTest {
     private void expectMockCONNECTClient() throws Exception {
         context.checking(new Expectations() {
             {
+                // TODO: Using "anything()" to match "Object..." due to JMock upgrade
                 oneOf(mockCONNECTClient).invokePort(with(any(Class.class)),
                         with(any(String.class)),
-                        with(any(RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType.class)));
+                        with(anything()));
                 will(returnValue(createValidRegistryResponse()));
             }
         });

--- a/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceUnsecuredImplTest.java
+++ b/Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceUnsecuredImplTest.java
@@ -105,9 +105,10 @@ public class EntityDocSubmissionProxyWebServiceUnsecuredImplTest {
     private void expectMockCONNECTClient() throws Exception {
         context.checking(new Expectations() {
             {
+                // TODO: Using "anything()" to match "Object..." due to JMock upgrade
                 oneOf(mockCONNECTClient).invokePort(with(any(Class.class)),
                         with(any(String.class)),
-                        with(any(RespondingGatewayProvideAndRegisterDocumentSetRequestType.class)));
+                        with(anything()));
                 will(returnValue(createValidRegistryResponse()));
             }
         });

--- a/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/PatientDiscoveryPolicyCheckerTest.java
+++ b/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/PatientDiscoveryPolicyCheckerTest.java
@@ -181,8 +181,8 @@ public class PatientDiscoveryPolicyCheckerTest {
 
         context.checking(new Expectations() {
             {
-                oneOf(mockPolicyEngineProxy).checkPolicy(with(any(CheckPolicyRequestType.class)),
-                        with(any(AssertionType.class)));
+                oneOf(mockPolicyEngineProxy).checkPolicy(with(aNull(CheckPolicyRequestType.class)),
+                        with(aNull(AssertionType.class)));
                 will(returnValue(response));
             }
         });
@@ -215,8 +215,8 @@ public class PatientDiscoveryPolicyCheckerTest {
 
         context.checking(new Expectations() {
             {
-                oneOf(mockPolicyEngineProxy).checkPolicy(with(any(CheckPolicyRequestType.class)),
-                        with(any(AssertionType.class)));
+                oneOf(mockPolicyEngineProxy).checkPolicy(with(aNull(CheckPolicyRequestType.class)),
+                        with(aNull(AssertionType.class)));
                 will(returnValue(response));
             }
         });

--- a/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/aspect/PRPAIN201306UV02EventDescriptionBuilderTest.java
+++ b/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/aspect/PRPAIN201306UV02EventDescriptionBuilderTest.java
@@ -34,6 +34,7 @@ import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.util.Base64Coder;
 import java.io.Serializable;
 import java.util.List;
+import static org.hamcrest.CoreMatchers.hasItems;
 import org.hl7.v3.BinaryDataEncoding;
 import org.hl7.v3.COCTMT090003UV01AssignedEntity;
 import org.hl7.v3.EDExplicit;
@@ -47,6 +48,7 @@ import org.hl7.v3.PRPAIN201306UV02MFMIMT700711UV01RegistrationEvent;
 import org.hl7.v3.PRPAIN201306UV02MFMIMT700711UV01Subject1;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 import org.junit.Before;
 import org.junit.Test;
@@ -178,8 +180,8 @@ public class PRPAIN201306UV02EventDescriptionBuilderTest extends BaseDescription
 
         EventDescription eventDescription = getEventDescriptionWithIIs(ii1, ii2);
         assertEquals(2, eventDescription.getRespondingHCIDs().size());
-        assertEquals(NhincConstants.HCID_PREFIX + "1.2.3", eventDescription.getRespondingHCIDs().get(0));
-        assertEquals(NhincConstants.HCID_PREFIX + "1.2.3.4", eventDescription.getRespondingHCIDs().get(1));
+        assertThat(eventDescription.getRespondingHCIDs(),
+            hasItems(NhincConstants.HCID_PREFIX + "1.2.3", NhincConstants.HCID_PREFIX + "1.2.3.4"));
     }
 
     private EventDescription getEventDescriptionWithIIs(II... iis) {

--- a/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/entity/proxy/EntityPatientDiscoveryProxyWebServiceSecuredImplTest.java
+++ b/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/entity/proxy/EntityPatientDiscoveryProxyWebServiceSecuredImplTest.java
@@ -183,8 +183,9 @@ public class EntityPatientDiscoveryProxyWebServiceSecuredImplTest {
                     .mock(RespondingGatewayPRPAIN201306UV02ResponseType.class);
             context.checking(new Expectations() {
                 {
+                    // TODO: Using "anything()" to match "Object..." due to JMock upgrade
                     atLeast(1).of(mockCONNECTClient).invokePort(with(any(Class.class)), with(any(String.class)),
-                            with(any(RespondingGatewayPRPAIN201305UV02RequestType.class)));
+                        with(anything()));
                     will(returnValue(mockResponse));
                 }
             });

--- a/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/entity/proxy/EntityPatientDiscoveryProxyWebServiceUnsecuredImplTest.java
+++ b/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/entity/proxy/EntityPatientDiscoveryProxyWebServiceUnsecuredImplTest.java
@@ -187,8 +187,9 @@ public class EntityPatientDiscoveryProxyWebServiceUnsecuredImplTest {
                     .mock(RespondingGatewayPRPAIN201306UV02ResponseType.class);
             context.checking(new Expectations() {
                 {
+                    // TODO: Using "anything()" to match "Object..." due to JMock upgrade
                     atLeast(1).of(mockCONNECTClient).invokePort(with(any(Class.class)), with(any(String.class)),
-                            with(any(EntityPatientDiscoveryPortType.class)));
+                            with(anything()));
                     will(returnValue(mockResponse));
                 }
             });

--- a/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/response/VerifyModeTest.java
+++ b/Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/response/VerifyModeTest.java
@@ -174,8 +174,8 @@ public class VerifyModeTest {
 
         context.checking(new Expectations() {
             {
-                exactly(1).of(mockTrustMode).processResponse(with(any(PRPAIN201306UV02.class)),
-                    with(any(AssertionType.class)), with(any(II.class)));
+                exactly(1).of(mockTrustMode).processResponse(with(aNull(PRPAIN201306UV02.class)),
+                    with(aNull(AssertionType.class)), with(any(II.class)));
             }
         });
 

--- a/Product/Production/Services/SharedEngineCore/src/test/java/gov/hhs/fha/nhinc/redactionengine/adapter/DocQueryResponseProcessorTest.java
+++ b/Product/Production/Services/SharedEngineCore/src/test/java/gov/hhs/fha/nhinc/redactionengine/adapter/DocQueryResponseProcessorTest.java
@@ -1394,8 +1394,8 @@ public class DocQueryResponseProcessorTest {
             };
             context.checking(new Expectations() {
                 {
-                    one(mockPatientConsentHelper).retrievePatientConsentbyDocumentId(with(any(String.class)),
-                            with(any(String.class)), with(any(String.class)));
+                    oneOf(mockPatientConsentHelper).retrievePatientConsentbyDocumentId(with(aNull(String.class)),
+                        with(aNull(String.class)), with(any(String.class)));
                 }
             });
             PatientPreferencesType patientPreferences = processor.retrievePatientPreferencesForDocument(extObject);

--- a/Product/Production/Services/SharedEngineCore/src/test/java/gov/hhs/fha/nhinc/redactionengine/adapter/DocRetrieveResponseProcessorTest.java
+++ b/Product/Production/Services/SharedEngineCore/src/test/java/gov/hhs/fha/nhinc/redactionengine/adapter/DocRetrieveResponseProcessorTest.java
@@ -77,8 +77,8 @@ public class DocRetrieveResponseProcessorTest {
             RetrieveDocumentSetResponseType retrieveResponse = new RetrieveDocumentSetResponseType();
             context.checking(new Expectations() {
                 {
-                    allowing(mockPatientConsentHelper).retrievePatientConsentbyDocumentId(with(any(String.class)),
-                            with(any(String.class)), with(any(String.class)));
+                    allowing(mockPatientConsentHelper).retrievePatientConsentbyDocumentId(with(aNull(String.class)),
+                        with(aNull(String.class)), with(aNull(String.class)));
                 }
             });
 

--- a/Product/Production/pom.xml
+++ b/Product/Production/pom.xml
@@ -89,7 +89,7 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
-                    <argLine>-XX:MaxPermSize=1012m -XX:-UseSplitVerifier</argLine>
+                    <argLine>-XX:MaxPermSize=1012m -XX:-UseSplitVerifier -noverify</argLine>
                     <systemProperties>
                         <property>
                             <name>com.sun.aas.instanceRoot</name>

--- a/Product/pom.xml
+++ b/Product/pom.xml
@@ -559,7 +559,7 @@
         <groupId>org.javassist</groupId>
         <artifactId>javassist</artifactId>
         <version>3.20.0-GA</version>
-        <scope>provided</scope>
+        <scope>runtime</scope>
       </dependency>
 
       <!-- Logging -->

--- a/Product/pom.xml
+++ b/Product/pom.xml
@@ -556,9 +556,9 @@
         <scope>provided</scope>
       </dependency>
       <dependency>
-        <groupId>javassist</groupId>
+        <groupId>org.javassist</groupId>
         <artifactId>javassist</artifactId>
-        <version>3.4.GA</version>
+        <version>3.20.0-GA</version>
         <scope>provided</scope>
       </dependency>
 
@@ -1087,25 +1087,25 @@
       <dependency>
         <groupId>org.jmock</groupId>
         <artifactId>jmock</artifactId>
-        <version>2.5.1</version>
+        <version>2.8.2</version>
         <scope>test</scope>
       </dependency>
       <dependency>
         <groupId>org.jmock</groupId>
         <artifactId>jmock-legacy</artifactId>
-        <version>2.5.1</version>
+        <version>2.8.2</version>
         <scope>test</scope>
       </dependency>
       <dependency>
         <groupId>org.jmock</groupId>
         <artifactId>jmock-junit4</artifactId>
-        <version>2.5.1</version>
+        <version>2.8.2</version>
         <scope>test</scope>
       </dependency>
       <dependency>
         <groupId>org.mockito</groupId>
         <artifactId>mockito-all</artifactId>
-        <version>1.9.0</version>
+        <version>1.10.19</version>
         <scope>test</scope>
       </dependency>
       <dependency>
@@ -1297,4 +1297,3 @@
     </profile>
   </profiles>
 </project>
-

